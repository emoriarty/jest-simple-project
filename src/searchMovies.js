const movies = [{
  title: 'Alien',
  year: '1979',
  rating: '★★★★★'
} , {
  title: 'Aliens',
  year: '1986',
  rating: '★★★★★'
} , {
  title: 'Alien 3',
  year: '1992',
  rating: '★★'
} , {
  title: 'Alien 4',
  year: '1979',
  rating: '★★'
} , {
  title: 'Prometheus',
  year: '2012',
  rating: '★'
} , {
  title: 'Alien: Covenant',
  year: '2017',
  rating: '★'
}]

const searchByTitle = query =>
  movies.filter(movie =>
    movie.title.toLowerCase().includes(query.toLowerCase()));

export default {
  byTitle: searchByTitle
}
