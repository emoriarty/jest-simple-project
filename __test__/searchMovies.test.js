import searchMovies from '../src/searchMovies';

describe('testing searchMovies function', () => {
  // Test against stored output
  test('by title', () =>
    expect(searchMovies.byTitle('prometheus')).toMatchSnapshot())

  test('is case insensitive', () =>
    expect(searchMovies.byTitle('prometheus').length).toBe(1))

  test('sequencial search', () =>
    ['a', 'alien', 'alien 3'].forEach((query) =>
      expect(searchMovies.byTitle(query)).toMatchSnapshot()
    )
  )
})
