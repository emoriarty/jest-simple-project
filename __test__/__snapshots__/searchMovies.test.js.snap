// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing searchMovies function by title 1`] = `
Array [
  Object {
    "rating": "★",
    "title": "Prometheus",
    "year": "2012",
  },
]
`;

exports[`testing searchMovies function sequencial search 1`] = `
Array [
  Object {
    "rating": "★★★★★",
    "title": "Alien",
    "year": "1979",
  },
  Object {
    "rating": "★★★★★",
    "title": "Aliens",
    "year": "1986",
  },
  Object {
    "rating": "★★",
    "title": "Alien 3",
    "year": "1992",
  },
  Object {
    "rating": "★★",
    "title": "Alien 4",
    "year": "1979",
  },
  Object {
    "rating": "★",
    "title": "Alien: Covenant",
    "year": "2017",
  },
]
`;

exports[`testing searchMovies function sequencial search 2`] = `
Array [
  Object {
    "rating": "★★★★★",
    "title": "Alien",
    "year": "1979",
  },
  Object {
    "rating": "★★★★★",
    "title": "Aliens",
    "year": "1986",
  },
  Object {
    "rating": "★★",
    "title": "Alien 3",
    "year": "1992",
  },
  Object {
    "rating": "★★",
    "title": "Alien 4",
    "year": "1979",
  },
  Object {
    "rating": "★",
    "title": "Alien: Covenant",
    "year": "2017",
  },
]
`;

exports[`testing searchMovies function sequencial search 3`] = `
Array [
  Object {
    "rating": "★★",
    "title": "Alien 3",
    "year": "1992",
  },
]
`;
